OpenRA.Mods.Common/Traits/Conditions/GrantConditionOnArmamentReloading.cs

```diff
@@ -0,0 +1,83 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2018 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Linq;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	public class GrantConditionOnArmamentReloadingInfo : ConditionalTraitInfo, Requires<ArmamentInfo>
+	{
+		[FieldLoader.Require]
+		[GrantedConditionReference]
+		[Desc("Condition to grant.")]
+		public readonly string Condition = null;
+
+		[Desc("Armament name.")]
+		public readonly string Armament = "primary";
+
+		public override object Create(ActorInitializer init) { return new GrantConditionOnArmamentReloading(init.Self, this); }
+	}
+
+	public class GrantConditionOnArmamentReloading : ConditionalTrait<GrantConditionOnArmamentReloadingInfo>, INotifyArmamentReloading
+	{
+		readonly Armament armament;
+
+		ConditionManager conditionManager;
+		int conditionToken = ConditionManager.InvalidConditionToken;
+
+		public GrantConditionOnArmamentReloading(Actor self, GrantConditionOnArmamentReloadingInfo info)
+			: base(info)
+		{
+			armament = self.TraitsImplementing<Armament>()
+				.Single(a => a.Info.Name == Info.Armament);
```

Comment:
Do we have an official policy on this one? How should eliteweapons be treated in this case? How are they treated in TS?

Reply:
> How should eliteweapons be treated in this case?

Elite weapons should have a different name than their non-elite counterpart, then you can use a 2nd GCOAR for the elite weapon.

@pchote said months ago that we should ultimately aim for one attack trait per armament, so each armament having a distinct name will become mandatory anyway, eventually.
I'm trying to write anything somewhat related with that in mind already, to keep the number of places that need to be adapted later as low as possible.

> How are they treated in TS?

Probably still using "primary" for both, but iirc there aren't any actors in TS that have both an elite weapon *and* one of the affected traits, so at this point it shouldn't matter.

Reply:
> @pchote said months ago that we should ultimately aim for one attack trait per armament, so each armament having a distinct name will become mandatory anyway, eventually.

Hmm, I don't remember this.

Reply:
Ok... anyway, I think the approach of this PR is much more robust and less issue-prone than attempting to control every Armament via a single condition trait would be.

However, it just occured to me that maybe making the Armament itself grant a `ReloadingCondition` would be even better. What do you think?